src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -539,6 +554,26 @@ public ErrorDiagnostic ArgumentCountMismatch(int argumentCount, int mininumArgum
                 TextSpan,
                 "BCP099",
                 $"The \"{LanguageConstants.ParameterAllowedPropertyName}\" array must contain one or more items.");
+
+            public ErrorDiagnostic IfFunctionNotSupported() => new ErrorDiagnostic(
+                TextSpan,
+                "BCP100",
+                "The \"if\" function is not supported. Use the ternary conditional operator instead.");

```

Comment:
Worth also giving an example for anyone not familiar with ternary operators?

Reply:
I'm guessing you're thinking of adding something like "<condition> ? <true value> : <false value>" to the message?

If you don't know how to use the ternary or have never seen it, I don't know if that'll help tbh. I think we're going to need to start adding aka.ms doc links to our error messages for some things. We could have an aka.ms/bicep/ternary link here that would answer all questions. I'll open an issue.

Reply:
Opened #681.